dlt/extract/extract.py

```diff
@@ -170,6 +175,64 @@ def __init__(
         self.original_data: Any = original_data
         super().__init__()
 
+    def _compute_metrics(self, load_id: str, source: DltSource) -> ExtractMetrics:
+        # map by job id
+        job_metrics = {
+            ParsedLoadJobFileName.parse(m.file_path): m
+            for m in self.extract_storage.closed_files(load_id)
+        }
+        # aggregate by table name
+        table_metrics = {
+            table_name: sum(map(lambda pair: pair[1], metrics), EMPTY_DATA_WRITER_METRICS)
+            for table_name, metrics in itertools.groupby(
+                job_metrics.items(), lambda pair: pair[0].table_name
+            )
+        }
+        # aggregate by resource name
+        resource_metrics = {
+            resource_name: sum(map(lambda pair: pair[1], metrics), EMPTY_DATA_WRITER_METRICS)
+            for resource_name, metrics in itertools.groupby(
+                table_metrics.items(), lambda pair: source.schema.get_table(pair[0])["resource"]
+            )
+        }
+        # collect resource hints
+        clean_hints: Dict[str, Dict[str, Any]] = {}
+        for resource in source.selected_resources.values():
+            # cleanup the hints
+            hints = clean_hints[resource.name] = {}
+            resource_hints = resource._hints or resource.compute_table_schema()
+
+            for name, hint in resource_hints.items():
+                if hint is None or name in ["validator"]:
+                    continue
+                if name == "incremental":
+                    # represent incremental as dictionary (it derives from BaseConfiguration)
+                    hints[name] = dict(hint)  # type: ignore[call-overload]
+                    continue
+                if name == "original_columns":
+                    # this is original type of the columns ie. Pydantic model
+                    hints[name] = get_full_class_name(hint)
+                    continue
+                if callable(hint):
+                    hints[name] = get_callable_name(hint)
+                    continue
+                if name == "columns":
+                    if hint:
+                        hints[name] = yaml.dump(
+                            hint, allow_unicode=True, default_flow_style=False, sort_keys=False
+                        )
+                    continue
+                hints[name] = hint
+
+        return {
+            "schema_name": source.schema.name,
+            "job_metrics": {job.job_id(): metrics for job, metrics in job_metrics.items()},
+            "table_metrics": table_metrics,
+            "resource_metrics": resource_metrics,
+            "dag": source.resources.selected_dag,
```

Comment:
so from this we can see which resources where selected during pipeline run, right? But the info which table stems from which resource is not in here, is it?

Reply:
you can get this information from the schema (there's `resource` tag for each table). also it won't be too hard to add this info.
I'd merge the stuff as it is. or make a clear request that you want it :) @sh-rp